### ISSUE TITLE
Build failing locally

### ISSUE THREAD

**Issue by d47853** [gchq/koryphe#192]
A false positive result in the CI resulted in a breaking build being released in 1.10.0. It looks like the binary operator tests fail as a result of tests being run in parallel and re-using the same variable. All the other (FunctionTests and PredicateTests) are not running at all:
```
...
Running uk.gov.gchq.koryphe.predicate.AdaptedPredicateTest
Tests run: 0, Failures: 0, Errors: 0, Skipped: 0, Time elapsed: 0 sec
Running uk.gov.gchq.koryphe.predicate.PredicateMapTest
Tests run: 0, Failures: 0, Errors: 0, Skipped: 0, Time elapsed: 0 sec

Results :

Failed tests:   uk.gov.gchq.koryphe.impl.binaryoperator.ProductTest.testAggregateInIntModeMixedInput(): java.lang.Short cannot be cast to java.lang.Integer
  uk.gov.gchq.koryphe.impl.binaryoperator.ProductTest.testAggregateInFloatMode(): java.lang.Long cannot be cast to java.lang.Float
...
```

**Comment by macenturalxl1** [comment]
Thank you for flagging this - I have just ran the `mvn clean install` command and see the compiler errors - will fix this now

**Comment by d47853** [comment]
Interestingly, it passes with -Ptravis which configures the surefire maven plugin. It appears the tests were running and passing correctly in CI but not locally where the surefire-maven-plugin is not configured.

**Comment by macenturalxl1** [comment]
I found it really strange to that sometimes it works / sometimes it doesn't with various commands (even the -q quiet argument sometimes hides logs when it should output a test has actually been ran) - which all in it self is a red flag as it should just work everytime.

I'll keep looking at the original error above when running install as a starting point and then look how it is breaking with the other commands or mvn/surefire configs....

**Comment by macenturalxl1** [comment]
Hi, figured it out - I've explicitly added the surefire-plugin 2.22.1 to <build /> so when I run `mvn clean install` it is picking up the correct version (as by default it picks up an older version not compatible with JUnit5).

I've raised a pull request - let me know what you think, thanks

**Comment by d47853** [comment]
Merged into develop
